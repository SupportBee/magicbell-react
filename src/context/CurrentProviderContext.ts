import { createContext } from 'react';

export const NO_PROVIDER = 'MAGICBELL_NO_PROVIDER';

const CurrentProviderContext = createContext<String>(NO_PROVIDER);
export default CurrentProviderContext;
