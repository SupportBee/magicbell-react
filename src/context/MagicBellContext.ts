import { createContext, useContext } from 'react';

export interface IMagicBellContext {
  images?: Partial<{
    emptyInboxUrl: string;
  }>;
}

const MagicBellContext = createContext<IMagicBellContext>({} as IMagicBellContext);
export default MagicBellContext;

const useMagicBellContext = () => useContext(MagicBellContext);
export { useMagicBellContext };
