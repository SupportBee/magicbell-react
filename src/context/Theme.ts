export interface IElementTheme {
  backgroundColor: string;
  backgroundOpacity: number;
  borderRadius: string;
  fontFamily: string;
  fontSize: string;
  textAlign: 'left' | 'right' | 'center' | 'justify' | 'initial' | 'inherit';
  textColor: string;
  textTransform: 'none' | 'capitalize' | 'uppercase' | 'lowercase' | 'initial' | 'inherit';
}

export interface IMagicBellTheme {
  icon: {
    borderColor: string;
    width: string;
  };
  unseenBadge: IElementTheme;
  header: IElementTheme;
  container: IElementTheme;
  footer: IElementTheme;
  notification: {
    default: IElementTheme;
    unread: IElementTheme;
    unseen: IElementTheme;
  };
}

const blue = '#3498F4';
const red = '#DF4759';
const black = '#3A424D';

// prettier-ignore
const fontFamily = '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen-Sans, Ubuntu, Cantarell, "Helvetica Neue", sans-serif';
const fontSize = '14px';
const textAlign = 'left';
const textTransform = 'none';

export const defaultTheme = {
  icon: {
    borderColor: blue,
    width: '24px',
  },
  header: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  footer: {
    backgroundColor: blue,
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: fontSize,
    textAlign,
    textColor: 'white',
    textTransform,
  },
  unseenBadge: {
    backgroundColor: red,
    backgroundOpacity: 1,
    borderRadius: '2px',
    fontFamily,
    fontSize: '10px',
    textAlign: 'center',
    textColor: 'white',
    textTransform,
  },
  container: {
    backgroundColor: '#FFFFFF',
    backgroundOpacity: 1,
    borderRadius: '8px',
    fontFamily,
    fontSize: '14px',
    textAlign,
    textColor: black,
    textTransform,
  },
  notification: {
    default: {
      backgroundColor: blue,
      backgroundOpacity: 0.1,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unread: {
      backgroundColor: blue,
      backgroundOpacity: 0,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
    unseen: {
      backgroundColor: blue,
      backgroundOpacity: 0.05,
      borderRadius: '8px',
      fontFamily,
      fontSize,
      textAlign,
      textColor: black,
      textTransform,
    },
  },
} as IMagicBellTheme;
