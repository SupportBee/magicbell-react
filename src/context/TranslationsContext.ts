import { createContext, useContext } from 'react';

const TranslationsContext = createContext<any>({});
export default TranslationsContext;

const TranslationsProvider = TranslationsContext.Provider;
const useTranslations = () => useContext(TranslationsContext);
export { TranslationsProvider, useTranslations };
