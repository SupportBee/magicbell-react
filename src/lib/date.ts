import dayjs from 'dayjs';
import localizedFormat from 'dayjs/plugin/localizedFormat';
import relativeTime from 'dayjs/plugin/relativeTime';
import updateLocale from 'dayjs/plugin/updateLocale';

dayjs.extend(localizedFormat);
dayjs.extend(relativeTime);
dayjs.extend(updateLocale);

dayjs.updateLocale('en', {
  relativeTime: {
    future: 'in %s',
    past: '%s ago',
    s: '%ds',
    m: '1m',
    mm: '%dm',
    h: '1h',
    hh: '%dh',
    d: '1d',
    dd: '%dd',
    M: '1mo',
    MM: '%dmo',
    y: '1y',
    yy: '%dy',
  },
});
