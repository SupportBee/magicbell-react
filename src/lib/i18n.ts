import dayjs from 'dayjs';
import 'dayjs/locale/es';
import 'dayjs/locale/pt-br';
import defaultsDeep from 'lodash/defaultsDeep';
import es from './translations/es';
import pt_br from './translations/pt-br';

export type CustomLocale = {
  name: string;
  translations: any;
};

function setDayjsLocale(localeName: string) {
  // All dayjs files are in lowercase and use hypens instead of underscores.
  const standardLocaleName = localeName.toLowerCase().replace('_', '-');
  dayjs.locale(standardLocaleName);
}

/**
 * Function to build localization data for the `TranslationsProvider`.
 *
 * @param locale
 */
export function useLocale(locale: string | CustomLocale) {
  const localeName = typeof locale === 'string' ? locale : locale.name;
  setDayjsLocale(localeName);

  const defaultLocales = { es, pt_BR: pt_br };

  if (typeof locale === 'string') return defaultLocales[locale];
  else {
    let customLocales = {};
    defaultsDeep(customLocales, { [localeName]: locale.translations }, defaultLocales);
    return customLocales[localeName];
  }
}
