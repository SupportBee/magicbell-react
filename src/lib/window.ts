/* eslint-disable no-restricted-globals */
import { useEffect, useRef, useState } from 'react';
import ResizeObserver from 'resize-observer-polyfill';

/**
 * Open a centered popup window.
 *
 * @param url Url of the floating window
 */
export function openWindow(url: string) {
  const top = (screen.height - 400) / 4;
  const left = (screen.width - 600) / 2;
  window.open(url, '', `width=600,height=400,scrollbars=no,top=${top},left=${left}`);
}

/**
 * Custom hook to observe the height of an HTML element.
 *
 * @param ref
 * @param initialHeight
 */
export function useHeight(ref, initialHeight) {
  const [height, setHeight] = useState(initialHeight);
  const resizeObserverRef = useRef() as any;

  useEffect(() => {
    resizeObserverRef.current = new ResizeObserver((entries: any = []) => {
      entries.forEach((entry) => {
        setHeight(entry.contentRect.height);
      });
    });

    if (ref.current) resizeObserverRef.current.observe(ref.current);

    return () => {
      if (resizeObserverRef.current) resizeObserverRef.current.disconnect();
    };
  }, [ref]);

  return height;
}
