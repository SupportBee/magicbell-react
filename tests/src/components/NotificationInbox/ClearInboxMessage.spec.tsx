import { render, RenderResult } from '@testing-library/react';
import React from 'react';
import ClearInboxMessage from '../../../../src/components/NotificationInbox/ClearInboxMessage';
import { MagicBellThemeProvider } from '../../../../src/context/MagicBellThemeContext';
import { defaultTheme } from '../../../../src/context/Theme';
import { TranslationsProvider } from '../../../../src/context/TranslationsContext';
import { useLocale } from '../../../../src/lib/i18n';

describe('components', () => {
  describe('ClearInboxMessage', () => {
    let view: RenderResult;

    beforeEach(() => {
      const translations = useLocale('es');

      view = render(
        <TranslationsProvider value={translations}>
          <MagicBellThemeProvider value={defaultTheme}>
            <ClearInboxMessage />
          </MagicBellThemeProvider>
        </TranslationsProvider>,
      );
    });

    afterEach(() => {
      view.unmount();
    });

    describe('render', () => {
      it('renders an image', () => {
        expect(view.container).toMatchSnapshot();
      });

      describe('default language', () => {
        it('renders the text in English', () => {
          view.rerender(
            <MagicBellThemeProvider value={defaultTheme}>
              <ClearInboxMessage />
            </MagicBellThemeProvider>,
          );

          expect(view.container).toMatchSnapshot();
        });
      });
    });
  });
});
