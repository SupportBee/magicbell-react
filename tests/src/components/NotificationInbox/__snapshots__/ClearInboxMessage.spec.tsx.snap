// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ClearInboxMessage render default language renders the text in English 1`] = `
.emotion-0 {
  padding: 3em!important;
  text-align: center;
  color: #3A424D;
}

.emotion-0 img {
  margin: 2em auto;
  height: 200px;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      <span>
        All clear!
        <br />
        We'll let you know when there's more.
      </span>
    </p>
    <img
      alt="No notifications"
      src="https://assets.magicbell.io/images/empty_inbox.png"
    />
  </div>
</div>
`;

exports[`components ClearInboxMessage render renders an image 1`] = `
.emotion-0 {
  padding: 3em!important;
  text-align: center;
  color: #3A424D;
}

.emotion-0 img {
  margin: 2em auto;
  height: 200px;
}

<div>
  <div
    class="emotion-0"
  >
    <p>
      <span>
        Â¡No tiene notificaciones!
        <br />
        Le haremos saber cuando llegue una.
      </span>
    </p>
    <img
      alt="No notifications"
      src="https://assets.magicbell.io/images/empty_inbox.png"
    />
  </div>
</div>
`;
